### ISSUE TITLE
install-poetry.py fails when using an empty PATH

### ISSUE THREAD

**Issue by dmarinuswoodwing** [python-poetry/install.python-poetry.org#43]
#### Description

We deploy poetry from cfn-init (which is a tool from AWS to run commands when an virtual machine is started). Apparently cfn-init starts processes with an (almost) empty environment.

#### Steps to reproduce

1. Start a shell with an empty environment:
    ```bash
    env -i bash
    ```

1. Execute the poetry installer:
    ```bash
    curl -sSL https://install.python-poetry.org/ | python3.7 - --version 1.2.0
    ```

#### Expected result

Either:

* Poetry properly installed (using a sane PATH) (preferred)
* A warning that you should configure your `PATH` 
* A warning that some command could not be executed.

#### Actual result:

```
Retrieving Poetry metadata

# Welcome to Poetry!

This will download and install the latest version of Poetry,
a dependency and package manager for Python.

It will add the `poetry` command to Poetry's bin directory, located at:

/root/.local/bin

You can uninstall at any time by executing this script with the --uninstall option,
and these changes will be reverted.

Installing Poetry (1.2.0): An error occurred. Removing partial environment.
Traceback (most recent call last):
  File "<stdin>", line 935, in <module>
  File "<stdin>", line 914, in main
  File "<stdin>", line 545, in run
  File "<stdin>", line 566, in install
  File "/usr/lib64/python3.7/contextlib.py", line 112, in __enter__
    return next(self.gen)
  File "<stdin>", line 638, in make_env
  File "<stdin>", line 624, in make_env
  File "<stdin>", line 314, in make
  File "/usr/lib64/python3.7/venv/__init__.py", line 66, in create
    self.setup_python(context)
  File "/usr/lib64/python3.7/venv/__init__.py", line 233, in setup_python
    copier(context.executable, path)
  File "/usr/lib64/python3.7/venv/__init__.py", line 176, in symlink_or_copy
    shutil.copyfile(src, dst)
  File "/usr/lib64/python3.7/shutil.py", line 120, in copyfile
    with open(src, 'rb') as fsrc:
FileNotFoundError: [Errno 2] No such file or directory: ''
```

**Comment by neersighted** [comment]
How realistic a concern is an empty `PATH`? Poetry is merely invoking the `venv` module, where the actual non-handling of this condition occurs.

While it is possible for us to prevent the condition (by detecting an empty `PATH` and trying to guess what a sane default or the OS path-helper is), or to try and detect this error (by combining a `FileNotFoundError` and the condition of `os.environ["PATH"]`), I really think this is either out of scope for Poetry, or an unreasonable expectation by the end user.

Shouldn't this be solved in `venv`, if it should be solved at all?

**Comment by meeuw** [comment]
> How realistic a concern is an empty `PATH`?

Like I reported, it happens with `cfn-init` and I ran into this problem.

I digged a bit futher and this is caused by `venv` trying to determine where the python executable is located by using `sys._base_executable`.
see: https://github.com/python/cpython/blob/16c33a9676e2f3ef330d09f2ab515c56636fa09f/Lib/venv/__init__.py#L130

The Poetry installer is recommended to be started without a full path so I presume Python uses `PATH` to determine which executable was started. If it cannot be found `sys._base_executable` is left empty and the installer fails.

This also means that setting `PATH` from the script doesn't help, I think the only solution would be to error out with a sane error message.

I will report a bug for the `venv` module but that wouldn't fix the Poetry installer.

Would it be acceptable to add:
```python
if not sys._base_executable:
    print("Couldn't determine base executable, please provide a PATH or provide the full path to the base executable (ie: /usr/bin/python)")
    sys.exit(1)
```
